src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildIntegrationTest.cs

```diff
@@ -325,6 +325,29 @@ public async Task Build_WithI8NOption_CopiesI8NAssembliesWithLinkerEnabled()
             Assert.FileDoesNotExist(result, buildOutputDirectory, "wwwroot", "_framework", "_bin", "I18N.West.dll");
         }
 
+        [Fact]
+        public async Task Build_WithCustomOutputPath_Works()
+        {
+            // Arrange
+            using var project = ProjectDirectory.Create("standalone", additionalProjects: new[] { "razorclasslibrary" });
+            project.AddDirectoryBuildContent(
+@"<PropertyGroup>
+    <BaseOutputPath>$(MSBuildThisFileDirectory)build\bin\</BaseOutputPath>
+    <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)build\obj\</BaseIntermediateOutputPath>
+</PropertyGroup>");
+
+            var result = await MSBuildProcessManager.DotnetMSBuild(project, args: "/restore");
```

Comment:
I slightly prefer setting this to passing it as a command line argument. With the project is edited, here's what a stitched together file would look like:

```xml
<!-- Blazor.props -->
<PropertyGroup>
  <_BlazorIntermediateOutputPath>$(BaseIntermediateOutputPath)blazor\</_BlazorIntermediateOutputPath>
</PropertyGroup>

<!-- Project.csproj -->
<PropertyGroup>
  <BaseIntermediateOutputPath>different-path</BaseIntermediateOutputPath>
</PropertyGroup>

<!-- Blazor.targets -->
```

Notice that this would still have a bug since `_BlazorIntermediateOutputPath` is calculated based on the initial value of  `BaseIntermediateOutputPath`. You wouldn't notice this if you pass `/p:BaseIntermediateOutputPath=...` since it changes default values for property group.

Reply:
OK cool